frontend/openchat-agent/src/utils/mapping.ts

```diff
@@ -5,9 +5,13 @@ export function optional<A, B>(candid: [] | [A], mapper: (a: A) => B): B | undef
     return candid[0] !== undefined ? mapper(candid[0]) : undefined;
 }
 
+export function optionalJson<A, B>(json: A | null, mapper: (a: A) => B): B | undefined {
+    return json != undefined ? mapper(json) : undefined;
```

Comment:
Weird that the input type is `A | null` and then we are checking if `json != undefined`